processing/src/test/java/org/apache/carbondata/processing/loading/partition/impl/RawRowComparatorTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.processing.loading.partition.impl;
+
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.carbondata.core.datastore.row.CarbonRow;
+import org.apache.carbondata.core.metadata.datatype.DataType;
+import org.apache.carbondata.core.metadata.datatype.DataTypes;
+import org.apache.carbondata.core.util.comparator.SerializableComparator;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class RawRowComparatorTest {
+  private RawRowComparator new_comparator;
+
+  @Test public void checkmap() {
+    DataType noDicDataTypes[] =
+        { DataTypes.INT, DataTypes.SHORT, DataTypes.BOOLEAN, DataTypes.BYTE, DataTypes.LONG };
+    boolean isSortColumnNoDict[] = { true, true, true, true, true };
+    int sortColumnIndices[] = { 1, 2, 3, 4, 5 };
+    new_comparator = new RawRowComparator(sortColumnIndices, isSortColumnNoDict, noDicDataTypes);
+    Field privateField = null;
+    try {
+      privateField = new_comparator.getClass().getDeclaredField("comparator_map");
+    } catch (NoSuchFieldException | SecurityException e) {
+      throw new RuntimeException(e);
+    }
+    privateField.setAccessible(true);
+    Map<DataType, SerializableComparator> testmap = new HashMap<>();
+    try {
+      Map<DataType, SerializableComparator> refMap =
+          (Map<DataType, SerializableComparator>) privateField.get(new_comparator);
+      Assert.assertTrue(refMap.size() == noDicDataTypes.length);
+    } catch (IllegalArgumentException | IllegalAccessException | SecurityException e) {
+      throw new RuntimeException(e);
+    }
+  }
+ 
+  @Test public void compareint() {
```

Comment:
Add a test case to check whether 2 RawRowComparator's use the same primitive comparator.

Reply:
Done